test/e2e/page-objects/pages/dialog/snap-interactive-dialog.ts

```diff
@@ -0,0 +1,97 @@
+import { Json } from '@metamask/utils';
+import { Driver } from '../../../webdriver/driver';
+import { strict as assert } from 'assert';
+
+const selectors = {
+  exampleDropdown: '[data-testid="snaps-dropdown"]',
+  exampleInput: '#example-input',
+  interactiveUITitle: {
+      text: 'Interactive UI Example Snap',
+      tag: 'p',
+    },
+  exampleSelectorDropdown: '[data-testid="snap-ui-renderer-card"]',
+  rendererPanel: '#snap-ui-renderer-panel',
+  selectorButton: '.mm-button-base--size-md',
+  exampleCheckbox: '.mm-checkbox__input',
+} satisfies Record<string, Json>
+
+class SnapInteractiveDialog {
+  private driver: Driver;
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+  }
+
+  async check_pageIsLoaded(): Promise<void> {
+    try {
+      await this.driver.waitForMultipleSelectors([
+        selectors.interactiveUITitle,
+        selectors.exampleInput,
+      ]);
+    } catch (e) {
```

Comment:
```suggestion
    } catch (e: string) {
```
Generally good practice to type errors, TS marks them as unknown here. If there was an API change and the above method changed it could lead to a runtime error.